File: SchoolERP.Server/Models/Leave & Gallery/gallery.cs
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SchoolERP.Server.Models.Leave___Gallery
{

    public class gallery
    {
    }

    [Table("gallery_category")]
    public class galleryCategory
    {
        [Key]
        public int category_id { get; set; }
        public string category_name { get; set; }
        public bool isActive { get; set; }
        public DateTime created_at { get; set; }
        public DateTime updated_at { get; set; }
        public string school_id { get; set; } 

        [ForeignKey("category_id")]
        public List<imageGallery>? imageGalleries { get; set; }
    }

    [Table("Image_gallery")]
    public class imageGallery
    {
        [Key]
        public int img_id { get; set; }
        public string img_url { get; set; }
        public int category_id { get; set; }
        public string? title { get; set; }
        public string? description { get; set; }
        public bool isActive { get; set; } 
        public DateTime uploaded_at { get; set; }
        public string school_id { get; set; }
    }

    public class imageGalleryDTOs
    {
        [Key]
        public int img_id { get; set; }
        public string img_url { get; set; }
        public int category_id { get; set; }
        public string title { get; set; }
        public string description { get; set; }
        public bool isActive { get; set; } 
        public DateTime uploaded_at { get; set; }
        public string school_id { get; set; }
        public IFormFile image { get; set; }
    }
}

File: schoolerp.client/src/app/models/gallery.ts

export interface holiday {
  id:number,
  title: string,
  description: string, 
  start_date:Date,
  end_date:Date,
  event_type: string,
  user_type: string,
  isDeleted: boolean,
  school_id: string,
  created_by: string,
  created_at: Date,
  updated_at: Date,
}

export interface galleryCategory {
  category_id:number,
  category_name: string,
  isActive: boolean,
  created_at:Date,
  updated_at:Date,
  school_id: string,
}

export interface imageGallery {
  img_id:number,
  img_url: string,
  category_id:number,
  title: string,
  description: string,
  isActive: boolean,
  uploaded_at: Date | string, 
  school_id: string,
}
 

export interface imageGImage extends imageGallery {
  image: File
}

